product-service/src/main/java/com/extremecoder/productservice/exception/GlobalExceptionHandler.java

```diff
@@ -0,0 +1,164 @@
+package com.extremecoder.productservice.exception;
+
+import lombok.extern.slf4j.Slf4j;
+import org.apache.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.http.converter.HttpMessageNotReadableException;
+import org.springframework.validation.BindException;
+import org.springframework.validation.FieldError;
+import org.springframework.validation.ObjectError;
+import org.springframework.web.HttpMediaTypeNotAcceptableException;
+import org.springframework.web.HttpMediaTypeNotSupportedException;
+import org.springframework.web.HttpRequestMethodNotSupportedException;
+import org.springframework.web.bind.MissingServletRequestParameterException;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.servlet.NoHandlerFoundException;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Global Exception Handler
+ *
+ * @author Badrul
+ */
+@Slf4j
+@ControllerAdvice
+public class GlobalExceptionHandler {
+
+    private static final String CONTENT_TYPE = HttpHeaders.CONTENT_TYPE;
+
+    /**
+     * HandleHttpMediaTypeError
+     *
+     * @param ex Exception
+     * @return Response Entity
+     */
+    @ExceptionHandler({HttpMediaTypeNotSupportedException.class, HttpMediaTypeNotAcceptableException.class})
+    public final ResponseEntity<Object> handleHttpMediaTypeError(Exception ex) {
```

Comment:
Please use `ResponseEntity<T>` instead of `ResponseEntity<Object>`

Reply:
We will update the response entity later

Reply:
ok